torch/_dynamo/variables/nn_module.py

```diff
@@ -211,35 +218,19 @@ def record_nn_module_stack():
                 )
 
             else:
-                # for lazy modules, run the pre-hooks which will update the type
-                # TODO mlazos: we don't fully support all of the hooks that exist,
-                # so restrict using __call__ only to lazy modules for now
                 assert self.source, (
                     "Must provide a valid source in order to inline, "
                     "since inlined function may have default args which must be guarded."
                 )
-                if is_lazy:
-                    if istype(mod.__call__, types.FunctionType):
-                        fn = mod.__call__
-                        fn_source = AttrSource(self.source, "__call__")
-                    else:
-                        assert istype(mod.__call__, types.MethodType)
-                        fn = mod.__call__.__func__
-                        fn_source = AttrSource(
-                            AttrSource(self.source, "__call__"), "__func__"
-                        )
-                        args = [self] + args
+                fn = mod.__call__
+                fn_source = AttrSource(self.source, "__call__")
```

Comment:
I remember we don't support nn Module only has ```__call__``` rather than ```forward``` function. Does it mean we will support it after this PR?

Reply:
yea, I think this PR will fix modules that only have __call__, at least in the majority of cases.

I am going to write another comment that raises a related question, specifically for the top-level module that you apply torch.compile() to.